### PR TITLE
[TS migration] MobileBudget tsx

### PR DESCRIPTION
**Pull request by joel-jeremy**

### CODE REVIEW

File: packages/loot-core/src/types/server-handlers.d.ts [hunk 1]
```diff
@@ -31,8 +34,6 @@ export interface ServerHandlers {
 
   'transaction-add': (transaction) => Promise<EmptyObject>;
 
-  'transaction-add': (transaction) => Promise<EmptyObject>;
-
```
Comment (MatissJanis) [inline packages/loot-core/src/types/server-handlers.d.ts:35]: **👏 praise:** ‏good catch

File: packages/loot-core/src/types/server-handlers.d.ts [hunk 2]
```diff
@@ -275,9 +273,9 @@ export interface ServerHandlers {
 
   'get-did-bootstrap': () => Promise<boolean>;
 
-  'subscribe-needs-bootstrap': (
-    args: { url } = {},
-  ) => Promise<
+  'subscribe-needs-bootstrap': (args: {
+    url;
+  }) => Promise<
```
Comment (MatissJanis) [inline packages/loot-core/src/types/server-handlers.d.ts:278]: **💭 thought:** ‏I wonder if there is some TS/eslint config we could enable to catch these.. 🤔

File: upcoming-release-notes/2081.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+category: Maintenance
+authors: [joel-jeremy]
+---
+
+MobileBudget tsx
```
Comment (MatissJanis) [inline upcoming-release-notes/2081.md:6]: **💬 suggestion:** ‏
```suggestion
Refactored MobileBudget component to TypeScript
```

**Review by MatissJanis — APPROVED** [review]
Thanks for doing the class -> hook refactor too!

**Review by MatissJanis — APPROVED** [review]

**Comment by joel-jeremy** [comment]
I'm holding off merging this until the split transaction and mobile category menu modals gets merged.

**Comment by MatissJanis** [comment]
Let me know when I should re-review this.

**Comment by joel-jeremy** [comment]
@MatissJanis This has been rebased and ready for review :)

**Comment by joel-jeremy** [comment]
Please merge the final no-default-export eslint PR before approving as I have rebased this from that branch.

File: packages/desktop-client/src/components/budget/MobileBudget.tsx [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+  const [bounds, setBounds] = useState({ start: currMonth, end: currMonth });
+  const [currentMonth, setCurrentMonth] = useState(currMonth);
+  const [initialized, setInitialized] = useState(false);
+  const [editMode, setEditMode] = useState(false);
+  const [editingBudgetCategoryId, setEditingBudgetCategoryId] = useState(null);
+  const [openBalanceActionMenuId, setOpenBalanceActionMenuId] = useState(null);
+
+  const loadCategories = async () => {
+    await props.getCategories();
+  };
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/MobileBudget.tsx:83]: **🥜 nitpick:** ‏no need to create a new function here. We can just use `props.getCategories` directly. Extra bonus: it maintains referential stability thus _should_ mean less re-renders.

**Review by MatissJanis — APPROVED** [review]

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**